core/src/main/scala/org/http4s/Entity.scala

```diff
@@ -16,19 +16,64 @@
 
 package org.http4s
 
-import cats.Monoid
+import fs2.{Chunk, Stream}
+import cats._
 import cats.syntax.all._
 
-final case class Entity[+F[_]](body: EntityBody[F], length: Option[Long] = None)
+sealed trait Entity[F[_]] {
+  def body: Stream[F, Byte]
+  def length: Option[Long]
+  def translate[G[_]](fk: F ~> G): Entity[G]
+  def covary[G[x] >: F[x]]: Entity[G] = translate[G](cats.arrow.FunctionK.id[F].widen[G])
+}
 
 object Entity {
+
+  case class Strict[F[_]](chunk: Chunk[Byte]) extends Entity[F] {
```

Comment:
Should these case classes be `final` or no?

Reply:
I think yes.  We made a pass through these with wartremover a long time ago, and I don't think any linter catches it now, and I usually forget myself.